src/main/java/com/gary/backendv2/service/IncidentService.java

```diff
@@ -0,0 +1,108 @@
+						.accidentReport(accidentReportService.getById(incident.getAccidentReport().getAccidentId()))
+						.build()
+			);
+		}
+		return incidentResponses;
+	}
+
+	public IncidentResponse getById(Integer id){
+		Optional<Incident> accidentReportOptional = incidentRepository.findByIncidentId(id);
+		if (accidentReportOptional.isEmpty()) throw new HttpException(HttpStatus.NOT_FOUND, String.format("Incident with id %s not found", id));
+		Incident incident = accidentReportOptional.get();
+
+		return IncidentResponse
+				.builder()
+				.incidentId(incident.getIncidentId())
+				.incidentStateType(incident.getIncidentStateType())
+				.dangerScale(incident.getDangerScale())
+				.reactionJustification(incident.getReactionJustification())
+				.accidentReport(accidentReportService.getById(incident.getAccidentReport().getAccidentId()))
+				.build();
+	}
+
+	public void addFromReport(AccidentReport accidentReport){
+		Incident incident = Incident
+				.builder()
+				.accidentReport(accidentReport)
+				.incidentStateType(IncidentStateType.NEW)
+				.build();
+		incidentRepository.save(incident);
+		accidentReport.setIncident(incident);
+		accidentReportRepository.save(accidentReport);
+	}
+
+	public void update (Integer id, IncidentRequest incidentRequest){
+		Optional<Incident> accidentReportOptional = incidentRepository.findByIncidentId(id);
+		if (accidentReportOptional.isEmpty()) throw new HttpException(HttpStatus.NOT_FOUND, String.format("Incident with id %s not found", id));
+		Incident incident = accidentReportOptional.get();
+
+		incident.setIncidentStateType(incidentRequest.getIncidentStateType());
+		incident.setDangerScale(incidentRequest.getDangerScale());
+		incident.setReactionJustification(incidentRequest.getReactionJustification());
+		incidentRepository.save(incident);
+	}
+
+	public void delete(Integer id){
+		Optional<Incident> accidentReportOptional = incidentRepository.findByIncidentId(id);
+		if (accidentReportOptional.isEmpty()) throw new HttpException(HttpStatus.NOT_FOUND, String.format("Incident with id %s not found", id));
+		incidentRepository.delete(accidentReportOptional.get());
+	}
+
+	public void addAmbulances(Integer id, List<String> ambulancesLicencePlates){
+		Optional<Incident> accidentReportOptional = incidentRepository.findByIncidentId(id);
+		if (accidentReportOptional.isEmpty()) throw new HttpException(HttpStatus.NOT_FOUND, String.format("Incident with id %s not found", id));
+		Incident incident = accidentReportOptional.get();
+		for (String s:ambulancesLicencePlates) {
+			Optional<Ambulance> ambulance = ambulanceRepository.findByLicensePlate(s);
+			if (ambulance.isEmpty()) throw new HttpException(HttpStatus.NOT_FOUND, String.format("Ambulance %s not found", s));
+			else {
+				Ambulance a = ambulance.get();
+				a.getIncidents().add(incident);
+				incident.getAmbulances().add(a);
+				ambulanceService.changeAmbulanceState(s, AmbulanceStateType.ON_ACTION);
+				ambulanceRepository.save(a);
+			}
+		}
```

Comment:
you can define method in a repository that takes Collection of Strings and returns Collection of Ambulances
you'll make one call to the database instead of N calls, where N is size of `ambulansLicensePlates` list

`List<Ambulance> selectedAmbulances = getAmbulancesByLicensePlateIsIn(List<String> licensePlates);`